### ISSUE TITLE
[BUG] 'Increment Version' workflow doesn't add new BWC Version for 2.0 and 2.x branches

### ISSUE THREAD

**Issue by tlfeng** [opensearch-project/OpenSearch#3218]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

**Comment by dblock** [comment]
Yeah, not sure it's worth fixing. https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/version.yml

**Comment by peterzhuamazon** [comment]
@tlfeng this would fail the main branch gradle check:
```
> Task :distribution:bwc:minor:buildBwcLinuxTar FAILED
 [2.1.0] 
 [2.1.0] FAILURE: Build failed with an exception.
 [2.1.0] 
 [2.1.0] * What went wrong:
 [2.1.0] Execution failed for task ':buildSrc:reaper:compileJava'.
 [2.1.0] > Could not create service of type UserHomeScopedCompileCaches using UserHomeScopeServices.createCompileCaches().
 [2.1.0]    > Timeout waiting to lock Java compile cache (/home/ec2-user/.gradle/caches/7.4.2/javaCompile). It is currently in use by another Gradle instance.
 [2.1.0]      Owner PID: 8658
 [2.1.0]      Our PID: 20754
 [2.1.0]      Owner Operation: 
 [2.1.0]      Our operation: 
 [2.1.0]      Lock file: /home/ec2-user/.gradle/caches/7.4.2/javaCompile/javaCompile.lock
 [2.1.0] 
 [2.1.0] * Try:
 [2.1.0] 2 actionable tasks: 2 executed
 [2.1.0] > Run with --stacktrace option to get the stack trace.
 [2.1.0] > Run with --info or --debug option to get more log output.
 [2.1.0] > Run with --scan to get full insights.
 [2.1.0] 
 [2.1.0] * Get more help at https://help.gradle.org
 [2.1.0] 
 [2.1.0] BUILD FAILED in 2m 9s
Output for /var/jenkins/workspace/gradle_check/search/distribution/bwc/minor/build/bwc/checkout-2.x/gradlew:

```

This does not happen in 2.0 which uses gradle 7.3.3, the issued main branch is using gradle 7.4.2.

Thanks.

**Comment by peterzhuamazon** [comment]
@reta ever since gradle upgrade to 7.4.2 I have seen such issue above on 3.0.0/main when running gradle check.
This does not happen in 2.0 branch with 7.3.3 gradle.

Running `./gradlew check -Dtests.coverage=true --no-daemon --no-scan`.

Thanks.

**Comment by CEHENKLE** [comment]
Hey @dreamer-89 Would you mind picking this up?  I understand this is one of the blocking issues to help get coverage out which I know we're all keen to see

**Comment by dreamer-89** [comment]
>Hey @dreamer-89 Would you mind picking this up? I understand this is one of the blocking issues to help get coverage out which I know we're all keen to see

Looking into this.

**Comment by dreamer-89** [comment]
> [2.1.0] > Could not create service of type UserHomeScopedCompileCaches using UserHomeScopeServices.createCompileCaches().
>  [2.1.0]    > Timeout waiting to lock Java compile cache (/home/ec2-user/.gradle/caches/7.4.2/javaCompile). It is currently in use by another Gradle instance.
>  [2.1.0]      Owner PID: 8658
>  [2.1.0]      Our PID: 20754
>  [2.1.0]      Owner Operation: 
>  [2.1.0]      Our operation: 
>  [2.1.0]      Lock file: /home/ec2-user/.gradle/caches/7.4.2/javaCompile/javaCompile.lock
>  [2.1.0]

This failure is different than the one this issue is intially reported (auto update Version.java for 2.x branches). From failure trace above, seems like lock contention among different gradle processes is causing the failure. 

This failure is not reproducible locally (ubuntu) with [latest changes](https://github.com/opensearch-project/OpenSearch/commit/304d830275fdbe5f8095a7285ca66266987c727a) from main; which makes me think, this is related to environment/setup related issue. This is coming from  https://github.com/opensearch-project/opensearch-build/pull/2249

**Comment by reta** [comment]
> Hi @reta ever since gradle upgrade to 7.4.2 I have seen such issue above on 3.0.0/main when running gradle check.
> This does not happen in 2.0 branch with 7.3.3 gradle.

I have not seen it yet, but I will take a look, thanks @dreamer-89 for troubleshooting

**Comment by reta** [comment]
Seems to be https://github.com/gradle/gradle/issues/20055

**Comment by peterzhuamazon** [comment]
@reta Update: resolved if you dont use AL2/CentOS but only use Ubuntu.

**Comment by dreamer-89** [comment]
> Seems to be [gradle/gradle#20055](https://github.com/gradle/gradle/issues/20055)

Seems happening on [AL2 specific instances](https://github.com/gradle/gradle/issues/20055#issuecomment-1099619015) that is what used here as well.

**Comment by reta** [comment]
Thanks a lot @peterzhuamazon !

**Comment by dreamer-89** [comment]
This issue appears during releases where a version bump on one branch fails the gradle check on all branches one version above.
